// dbr/index.js
var license = '';
var dbr = null;
if (process.platform === 'win32') {
    console.log('Windows');
    dbr = require('./dist/win/dbr');
    license = 't0068NQAAACqUjZa21C+W7fRdPkf2FRFRr+QpfVC2tDsl/8t25TzYCNxl5s0OkuwFgEMGNfN95Z0HYQ55ROi1px9JqVAP7/c=';
} else if (process.platform === 'linux') {
    console.log('Linux');
    dbr = require('./dist/linux/dbr');
    license = 't0068NQAAAHSLeRGgQNNSj5KP8wyR7gqp3AHhmaB9JY4fMDis6J7Veox1d6OJwzJK+pBqRVRcgRhbH8v04cy+KZ6o28jthlw=';
} else if (process.platform === 'darwin') {
    console.log('macOS');
    dbr = require('./dist/mac/dbr');
    license = 't0068NQAAAI3tqjND/UCGIOC+tXsEWOlmfWLbx1lXZhAUkuOj2lSxBo6Y+cjk+qxg7zYOGrPvE+u/z3ByXJRReIvN2VGr55Y=';
} else {
    console.log('Unknown Operating System');
}

var formats = {
    'OneD': 0x3FF,
    'PDF417': 0x2000000,
    'QRCode': 0x4000000,
    'DataMatrix': 0x8000000,
    'Aztec': 0x10000000
}

if (dbr) {
    var barcodeTypes = formats.OneD | formats.PDF417 | formats.QRCode | formats.DataMatrix | formats.Aztec;  // 1D, PDF417, QRCODE, DataMatrix, Aztec Code

    // Initialize DBR license.
    // Please visit https://www.dynamsoft.com/CustomerPortal/Portal/Triallicense.aspx or contact support@dynamsoft.com to get a valid trial or full license.
    dbr.initLicense(license);

    module.exports = {
        decodeFileAsync: dbr.decodeFileAsync,
        decodeFileStreamAsync: dbr.decodeFileStreamAsync,
        decodeBase64Async: dbr.decodeBase64Async,
        decodeYUYVAsync: dbr.decodeYUYVAsync,
        formats: formats,
        initLicense: dbr.initLicense,
        barcodeTypes: barcodeTypes
    };
}
